systemtest/src/test/java/io/strimzi/systemtest/log/LogSettingST.java

```diff
@@ -412,6 +417,39 @@ void testBridgeLogSetting(ExtensionContext extensionContext) {
         checkContainersHaveProcessOneAsTini(bridgeName);
     }
 
+    @ParallelTest
+    // This test might be flaky, as it gets real logs from CruiseControl pod
+    void testCruiseControlLogChange() {
+        final String debugText = " DEBUG ";
+        String cruiseControlPodName = PodUtils.getPodNameByPrefix(LOG_SETTING_CLUSTER_NAME + "-" + Constants.CRUISE_CONTROL_CONTAINER_NAME);
+        LOGGER.info("Check that default/actual root logging level is info");
+        String containerLogLevel = cmdKubeClient().execInPod(cruiseControlPodName, "grep", "-i", "rootlogger.level",
+                Constants.CRUISE_CONTROL_LOG_FILE_PATH).out().trim().split("=")[1];
+        assertThat(containerLogLevel.toUpperCase(Locale.ENGLISH), is(not(debugText.strip())));
+
+        LOGGER.info("Check logs in CruiseControl - make sure no DEBUG is found there.");
+        String logOut = cmdKubeClient().namespace(cluster.getNamespace()).execInCurrentNamespace("logs", cruiseControlPodName, "-c", Constants.CRUISE_CONTROL_CONTAINER_NAME, "--since=20s").out();
+        assertThat(logOut, not(containsString(debugText)));
+
+        InlineLogging logging = new InlineLogging();
+        logging.setLoggers(Collections.singletonMap("rootLogger.level", debugText.strip()));
+        KafkaResource.replaceKafkaResource(LOG_SETTING_CLUSTER_NAME, kafka -> kafka.getSpec().getCruiseControl().setLogging(logging));
+
+        try {
+            Thread.sleep(CO_OPERATION_TIMEOUT_SHORT);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+
+        LOGGER.info("Verify changed root logger in {}.", cruiseControlPodName);
+        containerLogLevel = cmdKubeClient().execInPod(cruiseControlPodName, "grep", "-i", "rootlogger.level", Constants.CRUISE_CONTROL_LOG_FILE_PATH).out().trim().split("=")[1];
+        assertThat(containerLogLevel.toUpperCase(Locale.ENGLISH), is(debugText.strip()));
```

Comment:
Wouldn't it be better to add this to `waitFor` and wait for `DEBUG` level to appear to the `rootlogger.level`? I guess it's better to wait for something to happen that just sleep for few seconds and then assert the result.

Reply:
(look at tests in `LoggingChangeST` where these waits are)

Reply:
Good idea, I will change it tomorrow.